osquery/tables/networking/windows/interfaces.cpp

```diff
@@ -19,88 +19,197 @@
 #include <boost/algorithm/string/join.hpp>
 
 #include <osquery/core.h>
+#include <osquery/logger.h>
 #include <osquery/tables.h>
 
+#include "osquery/core/conversions.h"
 #include "osquery/core/windows/wmi.h"
 
 namespace osquery {
 namespace tables {
 
-auto kMaxBufferAllocRetries = 3;
-auto kWorkingBufferSize = 15000;
-
-void genInterfaceDetail(const WmiResultItem& adapter, QueryData& results) {
-  Row r;
-  long lPlaceHolder;
-  bool bPlaceHolder;
-  std::vector<std::string> vPlaceHolder;
-  unsigned __int64 ulPlaceHolder;
-
-  adapter.GetString("MACAddress", r["mac"]);
-  adapter.GetString("AdapterType", r["type"]);
-  adapter.GetString("Description", r["description"]);
-  adapter.GetLong("InterfaceIndex", lPlaceHolder);
-  r["interface"] = SQL_TEXT(lPlaceHolder);
-  adapter.GetString("Manufacturer", r["manufacturer"]);
-  adapter.GetString("NetConnectionID", r["connection_id"]);
-  adapter.GetLong("NetConnectionStatus", lPlaceHolder);
-  r["connection_status"] = INTEGER(lPlaceHolder);
-  adapter.GetBool("NetEnabled", bPlaceHolder);
-  r["enabled"] = INTEGER(bPlaceHolder);
-  adapter.GetBool("PhysicalAdapter", bPlaceHolder);
-  r["physical_adapter"] = INTEGER(bPlaceHolder);
-  adapter.GetUnsignedLongLong("Speed", ulPlaceHolder);
-  r["speed"] = INTEGER(ulPlaceHolder);
+const auto kMaxBufferAllocRetries = 3;
+const auto kWorkingBufferSize = 15000;
+const auto kFreeMem = [](auto ptr) { free(ptr); };
```

Comment:
This is the default behavior for a smart pointer, you don't have to explicitly define and provide this lambda.